### DISCUSSION TITLE
Can I create pool by "ceph osd pool create test" in tool pods?

### DISCUSSION THREAD

**Discussion by mc956759** [rook/rook#12027] [Q&A]
I see "Rook allows creation and customization of storage pools through the custom resource definitions (CRDs)."
But the crd just support 'a-z'+'-' in yaml files.
So I want to create pool by use "ceph osd pool create" command.
What's different between these two ways? Could there be any potential problem?
Thanks for any help!

**Comment by galexrt** [comment]
Using the CRDs you just need to apply the "YAML" to your K8S cluster and the pool is created by the Rook Operator. The operator also just uses the ceph commands/APIs to create a pool.
The Rook Operator is bound to Kubernetes API "limitations" so that is why the pool name can't be more complex than `a-z+-`/ what "Kubernetes CRDs" naming schema allows.

There's not really a difference besides that you would need to manually run the commands to create the pools, though I would question if you really need more than `a-z-+` for a pool's name. Do you have an example for such a name?

**Reply by mc956759** [reply]
Thanks a lot! 
The pool-name is a problem left over from history. It's not convenient for us to change it.

**Comment by travisn** [comment]
@mc956759 You can also create a pool with any character even with the `.` or `_` if you specify the `spec.name` in the pool CR. See the example [here](https://github.com/rook/rook/blob/master/deploy/examples/pool-builtin-mgr.yaml#L11-L14).

**Reply by mc956759** [reply]
Thanks a lot! It's really helpful.
